web-registry/src/components/organisms/ChooseCreditClass/ChooseCreditClassLegacy.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { ChooseCreditClassGrid } from './ChooseCreditClass.Grid';
+import { ChooseCreditClassItem } from './ChooseCreditClass.Item';
+import { validate, getProjectPageBaseData } from '../../../lib/rdf';
+import {
+  useUpdateProjectByIdMutation,
+  useAllCreditClassesQuery,
+  useShaclGraphByUriQuery,
+} from '../../../generated/graphql';
+import { useErrorTimeout } from './hooks/useErrorTimeout';
+
+const ChooseCreditClassLegacy: React.FC = () => {
+  const navigate = useNavigate();
+  const { projectId } = useParams();
+  const [error, setError] = useErrorTimeout();
+  const { data: creditClassesData } = useAllCreditClassesQuery();
+  const { data: graphData } = useShaclGraphByUriQuery({
+    variables: {
+      uri: 'http://regen.network/ProjectPageShape',
+    },
+  });
+  const [updateProject] = useUpdateProjectByIdMutation();
+
+  async function handleSelection(
+    creditClassId: string,
+    creditClassUri: string,
+  ): Promise<void> {
+    if (graphData?.shaclGraphByUri?.graph) {
+      const metadata = getProjectPageBaseData();
+      metadata['regen:creditClass'] = {
+        '@type': 'regen:CreditClass',
+        '@id': creditClassUri,
+      };
+      const report = await validate(
+        graphData.shaclGraphByUri.graph,
+        metadata,
+        'http://regen.network/ProjectPageCreditClassGroup',
+      );
+      if (report.conforms) {
+        try {
+          await updateProject({
+            variables: {
+              input: {
+                id: projectId,
+                projectPatch: {
+                  creditClassId,
+                  metadata,
+                },
+              },
+            },
+          });
+          navigate(`/project-pages/${projectId}/basic-info`);
+        } catch (e) {
+          setError(`There was a problem updating this project: ${e}`);
+        }
+      }
+    }
```

Comment:
can we return early here and avoid some of the nesting?
```suggestion
      if (!report.conforms) {
        return
      }
      try {
        await updateProject({
          variables: {
```
etc

Reply:
👍